src/main/scala/util/RunUtils.scala

```diff
@@ -153,20 +160,58 @@ object RunUtils {
 
     val cfg = ProgramCfgFactory().fromIR(IRProgram)
 
+    val domain = computeDomain(IntraProcIRCursor, IRProgram.procedures)
+
     Logger.info("[!] Running Constant Propagation")
     val constPropSolver = ConstantPropagationSolver(cfg)
     val constPropResult = constPropSolver.analyze()
 
+    def newSolverTest(): Unit = {
+      val ilcpsolver = IRSimpleValueAnalysis.Solver(IRProgram)
+      val newCPResult = ilcpsolver.analyze()
+
+      val newCommandDomain = ilcpsolver.domain.filter(_.isInstanceOf[Command])
+
+      val newRes = newCPResult.flatMap((x, y) => y.flatMap {
+        case (_, el) if el == FlatLattice[BitVecLiteral].top || el == FlatLattice[BitVecLiteral].bottom => None
+        case z => Some(x -> z)
+      })
+      val oldRes = constPropResult.filter((x,y) => x.isInstanceOf[CfgNodeWithData[CFGPosition]]).flatMap((x, y) => y.flatMap {
+        case (_, el) if el == FlatLattice[BitVecLiteral].top || el == FlatLattice[BitVecLiteral].bottom => None
+        case z => Some(x.asInstanceOf[CfgNodeWithData[Any]].data -> z)
+      })
+      val both = newRes.toSet.intersect(oldRes.toSet)
+      val notnew = (newRes.toSet).filter(x => !both.contains(x)).toList.sorted((a, b) => a._2._1.name.compare(b._2._1.name))
+      val notOld = (oldRes.toSet).filter(x => !both.contains(x)).toList.sorted((a,b) => a._2._1.name.compare(b._2._1.name))
+      // newRes and oldRes should have value equality
+
+      //config.analysisResultsPath.foreach(s => writeToFile(printAnalysisResults(IRProgram, newCPResult), s"${s}_newconstprop$iteration.txt"))
+      config.analysisResultsPath.foreach(s => writeToFile(toDot(IRProgram), s"program.dot"))
+      config.analysisResultsPath.foreach(s => writeToFile(toDot(IRProgram, newCPResult.map((k,v) => (k, v.toString))), s"program-constprop.dot"))
+
+      config.analysisResultsPath.foreach(s => writeToFile(printAnalysisResults(IRProgram, newCPResult), s"${s}_new_cpres$iteration.txt"))
+      config.analysisResultsPath.foreach(s => writeToFile(printAnalysisResults(IRProgram, cfg, constPropResult), s"${s}_old_cpres$iteration.txt"))
+
+    }
+    newSolverTest()
```

Comment:
This should probably be in a test suite or something instead of being a core part of the analysis? Change that and then we can merge everything in.

Reply:
I've just removed this and made it run the analysis and output the results like the other constprop analysis. The results won't be the same anyway because this handles calls more carefully.

Reply:
That makes sense. What's the difference with handling calls, does it do some sort of havoc instead of just skipping over them in the intraprocedural analysis?